src/AppInstallerRepositoryCore/Public/winget/PackageTrackingCatalog.h

```diff
@@ -1,30 +1,37 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 #pragma once
-#include <Public/winget/RepositorySource.h>
+#include <AppInstallerStrings.h>
+#include <winget/Manifest.h>
 
 #include <memory>
 
 
 namespace AppInstaller::Repository
 {
+    struct Source;
+    struct SearchRequest;
+    struct SearchResult;
+    enum class PackageVersionMetadata;
+
     // A catalog for tracking package actions from a given source.
     struct PackageTrackingCatalog
     {
+        friend Source;
+
         PackageTrackingCatalog();
         PackageTrackingCatalog(const PackageTrackingCatalog&);
         PackageTrackingCatalog& operator=(const PackageTrackingCatalog&);
         PackageTrackingCatalog(PackageTrackingCatalog&&) noexcept;
         PackageTrackingCatalog& operator=(PackageTrackingCatalog&&) noexcept;
         ~PackageTrackingCatalog();
 
-        // Creates or opens the tracking catalog for the given source.
-        // TODO: Make creation exclusive to the refactored Source type.
-        static PackageTrackingCatalog CreateForSource(const Source& source);
-
-        // Removes the package tracking catalog for a given source.
+        // Removes the package tracking catalog for a given source identifier.
         static void RemoveForSource(const std::string& identifier);

```

Comment:
>RemoveForSource [](http://example.com/codeflow?start=20&length=15)

I saw this is only used in unit tests now, will there be usage in the product code in the future?

Reply:
Not sure, but since I wanted to use it in the tests I implemented it here.  We could call it as part of removing a source, but I wasn't sure if that was the right thing to do.